dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletFactory.java

```diff
@@ -292,22 +292,17 @@ protected abstract Contentlet findContentletByIdentifierAnyLanguage(String ident
 	 */
 	protected abstract List<Contentlet> findAllVersions(Identifier identifier, boolean bringOldVersions) throws DotDataException, DotSecurityException;
 
-	/**
-	 * Converts a "fat" (legacy) contentlet into a new contentlet.
-	 * @param Fat contentlet to be converted.
-	 * @return A "light" contentlet.
-	 * @throws DotDataException
-	 * @throws DotSecurityException 
-	 */
-	public abstract Contentlet convertFatContentletToContentlet (com.dotmarketing.portlets.contentlet.business.Contentlet fatty) throws DotDataException, DotSecurityException;
-	
-	/**
-	 * Converts a "light" contentlet into a "fat" (legacy) contentlet.
-	 * @param A "light" contentlet to be converted.
-	 * @return Fat contentlet.
-	 * @throws DotDataException
-	 */
-	public abstract com.dotmarketing.portlets.contentlet.business.Contentlet convertContentletToFatContentlet (Contentlet cont, com.dotmarketing.portlets.contentlet.business.Contentlet fatty) throws DotDataException;
+    /**
+     * Retrieves all versions for a contentlet identifier.
+     * @param identifier
+     * @param bringOldVersions Include old versions of contents, so it will return only live/working
+     * versions of contents, regardless of their languages
+     * @param maxResults
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public abstract List<Contentlet> findAllVersions(Identifier identifier, boolean bringOldVersions, final Optional<Integer> maxResults) throws DotDataException, DotSecurityException;
```

Comment:
Optional is not supposed to be used as parameter

have a look 
https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments
I think the way is  method overloading

Reply:
Done